src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -531,6 +541,18 @@ namespace xclcpuemhal2 {
           memTopology = new char[memTopologySize];
           memcpy(memTopology, xclbininmemory + sec->m_sectionOffset, memTopologySize);
         }
+        //Extract EMULATION_DATA from XCLBIN       
+        if (auto sec = xrt_core::xclbin::get_axlf_section(top, EMULATION_DATA)) {
+          emuDataSize = sec->m_sectionSize;
+          emuData = new char[emuDataSize];
```

Comment:
No raw new and delete, please.  use std::vector<char>

Reply:
Sure @stsoe , I need to make this kind of changes in other places of the code. I will implement those changes and submit it in other PR. Submitted changes were tested and also ran regression cases are passing.  So, please merge these changes.

Reply:
@stsoe , we need this raw pointer later for ofstream write.

Reply:
You never need raw pointers.  You can use std::vector or you can use std::unique_ptr<char[]>.

Reply:
@stsoe , removed usage of raw pointers and used std::unique_ptr<char[]> in the code. please review and merge.